### PR TITLE
Enable reverse op for complex values

### PR DESCRIPTION
**Pull request by ibab**
I've enabled complex values for the reverse op based on #3271.
This is part of the more general #2577.

I've also extended the tests to cover all enabled dtypes.
One thing that I've noticed is that we're compiling GPU code for `bool` and `int32`, but we're not enabling GPU kernels for these dtypes.
I've added commented out versions of the kernel registrations for these dtypes to point out that they're currently not enabled.

### CODE REVIEW

**Comment by vrv** [comment]
@girving can I assign this one to you?  If not, feel free to re-assign.

File: tensorflow/core/kernels/reverse_op.cc [hunk 1]
```diff
@@ -149,9 +145,13 @@ TF_CALL_double(DECLARE_GPU_SPEC);
                           ReverseOp<GPUDevice, T>)
 TF_CALL_uint8(REGISTER_GPU_KERNEL);
 TF_CALL_int8(REGISTER_GPU_KERNEL);
+//TF_CALL_int32(REGISTER_GPU_KERNEL);
+//TF_CALL_bool(REGISTER_GPU_KERNEL);
```
Comment (girving) [inline tensorflow/core/kernels/reverse_op.cc]: Why are these commented out?  If there are broken reasons why they can't be added, maybe leave them commented out but add a TODO comment saying why.

**Comment by girving** [comment]
Looks good except for the comment about the comment.

File: tensorflow/core/kernels/reverse_op.cc (hunk 1 above)
Comment (ibab) [inline tensorflow/core/kernels/reverse_op.cc]: I've tried enabling them, but had a problem with `int32`.
The tests were crashing without an error message for me :/
So I wasn't quite sure what to put in the TODO comment.
Should I add a comment saying `// TODO Find out why int32 doesn't work` ?

**Comment by ibab** [comment]
Okay, I've added a `TODO` to the commented out op registrations.
I could have a look at enabling those in a second PR.

**Comment by girving** [comment]
Jenkins, test this please.

**Comment by girving** [comment]
Jenkins, test this please.

**Comment by girving** [comment]
@vrv Do you know the tests wouldn't be triggering here?

**Comment by vrv** [comment]
i dunno, Jenkins is not exactly reliable ;).

@tensorflow-jenkins test this please

**Comment by girving** [comment]
Failures seem unrelated.  Merging.  Thank you for the stream of contributions!

**Pull request merged**